hbase-server/src/main/java/org/apache/hadoop/hbase/master/assignment/AssignmentManager.java

```diff
@@ -1859,11 +1859,17 @@ public void markRegionAsMerged(final RegionInfo child, final ServerName serverNa
       throws IOException {
     final RegionStateNode node = regionStates.getOrCreateRegionStateNode(child);
     node.setState(State.MERGED);
+    regionStateStore.mergeRegions(child, mergeParents, serverName);
+
+    // The order of adding merge qualifers and deleting from regionStates is important.
+    // hbck chore depends on these merge qualifers and regionStates to check if a region on FS
+    // is an orphan. If deleting from regionStates first, then there is a small gap that
+    // a region is not in regionStates and there is no merge qualifer for this region, it will be
+    // reported as an orphan.
     for (RegionInfo ri: mergeParents) {
       regionStates.deleteRegion(ri);
 
     }
-    regionStateStore.mergeRegions(child, mergeParents, serverName);
```

Comment:
ok. good. lets keep an eye on this one because sometimes fun, unexpected conditions when change order of operations.

Reply:
This happens during my test, which is doing a hbck report every 10 ms, yeah, will check the sequence to make sure it does not cause any unexpected conditions.

Reply:
I dropped this change as the fix is only targeting to a corner case for a small time window (~10ms). regionMap is being referenced in more than 100 places, it is hard to figure out every possible case.